CASMPET-6934 iSCSI SBPS: md doc for worker node(s) personalization

ravikanth-nalla-hpe opened the pull request:
…ation

<!---
    NOTE: This HTML style comment does not show in the PR.


    PULL-REQUESTS ARE LOOKED AT EVERY WORK DAY - WHEN THE MERGE BUTTON IS GREEN THE PR
    MAY BE MERGED.

    If more time is needed for review, please do one of the following:

    - Set the PR to a DRAFT; this allows reviewers to approve or request changes while disabling the merge button.
    - Prefix your PR title with "WIP" to indicate it is a "work in progress"; this is an indicator, it does not disable the merge button

    If neither of those items are present, then a pull-request is seen as a pull-request (e.g. a request to pull new content in) and it may
    be merged by a repository maintainer or CASM release manager at will.

--->
# Description
<!--- 
    NOTE: This HTML style comment does not show in the PR.

    Describe what this change is and what it is for.

    Include any related PRs URLs:

Relates to:
- pr-link-1
- pr-link-N
-->

# Checklist
<!---
    NOTE: This HTML style comment does not show in the PR. 

    An empty check is two brackets with a space in-between, a checked checkbox is two brackets with an x in-between

    unchecked checkbox: [ ]
    checked checkbox: [x]
    invalid checkbox: []
    invalid checkbox: [x ]
    invalid checkbox: [ x]
    invalid checkbox: [ x ]
-->

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

<!--- These are Markdown Reference Style URLs, they do not show in the PR --> 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+# iSCSI SBPS worker node(s) personalization
+As part of iSCSI SBPS solution, we need to setup/ configure worker nodes as iSCSI targets (servers) with necessary provisioning, configuration and enablement of required components. 
```
"part of iSCSI" -> "part of **the** iSCSI"

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+* enable spire for authentication used for IMS/ S3 access
+* create DNS SRV and A records in order to discover iSCSI targets from iSCSI initiators/clients (passed as boot parameter)
+* mount s3 images (boot-images bucket) with new, dedicated s3 user (ISCSI-SBPS) read only bucket policy via s3fs
+* apply k8s label (`iscsi=sbps`), to be consumed by Goss test suit and LIO Metrics for identification of nodes on which these has to run.
```
"consumed by Goss" -> "consumed by **the** Goss"
"these has to run" -> "these **have** to run"

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+* mount s3 images (boot-images bucket) with new, dedicated s3 user (ISCSI-SBPS) read only bucket policy via s3fs
+* apply k8s label (`iscsi=sbps`), to be consumed by Goss test suit and LIO Metrics for identification of nodes on which these has to run.
+
+Target worker node selection is via dynamic inventory stored in HSM (Hardware State Manager) – default to all worker nodes (`Management_Worker` group).
```
"in HSM" -> "in **the** HSM"
"default to all" -> "**the** default **is** all"

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
````diff
@@ -0,0 +1,250 @@
+```
+
+**Note: To delete any old/ stale session with similar name**
+If CFS session is already created and when you want to re create the new CFS session with same name for any reason 
````
"If CFS session" -> "If **a** CFS session"
"created and when" -> "created when"
"re create" -> "recreate"

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+
+**Note: To delete any old/ stale session with similar name**
+If CFS session is already created and when you want to re create the new CFS session with same name for any reason 
+it may not allow with error "Error: Conflicting session name: A session with the name iscsi-sbps-targets-config already exists".
```
"it may not allow with error" -> "it may not allow **it** with **the** error"

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
````diff
@@ -0,0 +1,250 @@
+```
+
+### **Track the Status of Ansible playbooks**
+Make sure that all the CFS play books have been completed successfully.
````
"have been completed" -> "have completed"

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+
+### **Track the Status of Ansible playbooks**
+Make sure that all the CFS play books have been completed successfully.
+Look for message "All playbooks completed successfully" in the cfs ansible container log.
```
"Look for message" -> "Look for **the** message"

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+### **Create CFS config file as below**
+See [CFS Configuration](CFS_Configurations.md) for more information.
+
+**Eg:**
```
"Eg" -> "E.g."

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+### **Create CFS config**
+See [CFS Configuration](CFS_Configurations.md) for more information.
+
+**Eg:**
```
"Eg" -> "E.g."

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+### **Create CFS session with CFS config**
+See [CFS_Sessions.md](CFS_Sessions.md) for more information.
+
+**Eg:**
```
"Eg" -> "E.g."

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+
+You can delete the old/ stale CFS session before creating the new CFS session again.
+
+**Eg:**
```
"Eg" -> "E.g."

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+### **View CFS session / check status**
+See [View CFS Sessions](CFS_Sessions.md) for more info.
+
+**Eg:**
```
"Eg" -> "E.g."

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
```diff
@@ -0,0 +1,250 @@
+
+See [Troubleshooting](Track_the_Status_of_a_Session.md) for more info.
+
+**Eg:**
```
"Eg" -> "E.g."

ravikanth-nalla-hpe commented:
The original Goss test link is going to be "https://github.com/Cray-HPE/sbps-marshal/GOSS_tests_for_sbps.md", which is currently "https://github.com/Cray-HPE/sbps-marshal/blob/CASMPET-7214-goss-sbps-doc/GOSS_tests_for_sbps.md" as mentioned in the md doc (under review: https://github.com/Cray-HPE/sbps-marshal/pull/12).

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
````diff
@@ -0,0 +1,251 @@
+```
+
+### **Clone VCS repo and check the latest `csm-config` version (Optional)**
+```bash
````
Take off the `bash`, this is causing the words `for` in lines 58-59 to get highlighted in red.  In general don't use ```bash unless what you are quoting is a bash script.  In this case it is an example output from running a command.

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Customization.md:
````diff
@@ -0,0 +1,251 @@
+ncn-m001:/home/csm-config-management #
+```
+
+```bash
````
Same here, remove `bash` as it is causing the `for` in line 235 and all of line 241 to get highlighted in red.  This is example output, not a bash script.  You should probably remove `bash` from all the comments, since none of them are actually quoting bash scripts.

dborman-hpe commented:
Please address the failing Markdown checks.  For example, there are a bunch of lines that end with a trailing space and some spelling errors.

dborman-hpe commented:
I checked this out with git to look at the file.  One thing that I noticed is that the file is full of "\r\n" line terminations (like in windows), instead of just "\n" which is normal for Linux.  Please fix that to be just "\n" instead of "\r\m".  One way to strip them out is to run something like:
```
# sed -e 's/\r//' < iSCSI_SBPS_Node_Personalization.md >xx
# mv xx iSCSI_SBPS_Node_Personalization.md
```

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Personalization.md:
````diff
@@ -0,0 +1,359 @@
+```
+
+#### **Check to see if the images/ LUN mapping created under "targetcli ls" cmd**
+Check to see if the “targetcli ls” output shows that fileio backing store are created for rootfs images along with corresponding iSCSI LUNs
````
It's subtle to see, but the double quotes in `“targetcli ls”` are fancy multi-character sequences, please change them to simple double quotes like `"targetcli ls"` (like they are on the previous line)

ravikanth-nalla-hpe replied:
Fixed

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Personalization.md:
```diff
@@ -0,0 +1,359 @@
+* mount s3 images (`boot-images` bucket) with new, dedicated s3 user (`ISCSI-SBPS`) read only bucket policy via s3fs
+* apply k8s label (`iscsi=sbps`), to be consumed by the Goss test suit and LIO Metrics for identification of nodes on which these have to run.
+
+Target worker node selection is via dynamic inventory stored in HSM (Hardware State Manager) – the default is all worker nodes (`Management_Worker` group).
```
Not easy to see, but in `Manager) – the` the dash is not a simple ascii dash character, please change it to `Manager) - the`

ravikanth-nalla-hpe replied:
Fixed

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Personalization.md:
```diff
@@ -0,0 +1,359 @@
+# iSCSI SBPS worker node(s) personalization
+
```
This whole file has `\r\n` line terminations, they should be just `\n`.  Please strip out all the `\r`s from this file.

leliasen-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Personalization.md:
```diff
@@ -0,0 +1,359 @@
+Ended sub-run, tag: hMUDa-vcs (duration: 3.982640351s)
+Ended run, tag: hMUDa (duration: 3.994254571s)
+SUCCESS: All 1 service tests passed: vcs
+ncn-m001:/home #
```
```suggestion

```
This is just my opinion and it works as is, but I would remove the prompt at the end of each block. I do not think you need to leave the `ncn-m001:/home #` at the end of the output. If you choose to remove it, be sure to remove the ending prompts in the rest of the document.

ravikanth-nalla-hpe replied:
Fixed

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Personalization.md:
```diff
@@ -0,0 +1,359 @@
+# iSCSI SBPS worker node(s) personalization
+
```
You can't see it, but this whole file has `\r\n` line terminations, they should be just `\n`.  Please strip out all the `\r`s from this file.

ravikanth-nalla-hpe replied:
Fixed

ravikanth-nalla-hpe commented:
> I checked this out with git to look at the file. One thing that I noticed is that the file is full of "\r\n" line terminations (like in windows), instead of just "\n" which is normal for Linux. Please fix that to be just "\n" instead of "\r\m". One way to strip them out is to run something like:
> 
> ```
> # sed -e 's/\r//' < iSCSI_SBPS_Node_Personalization.md >xx
> # mv xx iSCSI_SBPS_Node_Personalization.md
> ```

Fixed

ravikanth-nalla-hpe commented:
> Please address the failing Markdown checks. For example, there are a bunch of lines that end with a trailing space and some spelling errors.

I have checked and fixed all of the markdown check errors except the Goss md doc link
[1] Markdown link check:  The link provided is an official link and it is going to be available once 
PR https://github.com/Cray-HPE/sbps-marshal/pull/12 is merged to master branch of "sbps-marshal". So ignoring it as we discussed.
[2] Markdown spellcheck: spell errors taken care wherever necessary but there are 19 spell check errors reported still which are not actual spell errors. So ignoring it.
[3] Markdown style check: Fixed all the style check errors.

leliasen-hpe reviewed (approved):
Looks good to me except for the spellcheck issues. Those should be addressed. Words that are actually words like enablement should be added to the .spelling file in docs-csm. Some words might just need to be capitalized like Ansible and S3 . Others like rootfs might need to go in ``. You can search for these words in docs-csm and see how they are used in other documents.

ravikanth-nalla-hpe commented:
> Member
Fixed

leliasen-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Personalization.md:
````diff
@@ -0,0 +1,357 @@
+...
+```
+
+#### Check to see if the images/ LUN mapping created under `targetcli ls` cmd
````
```suggestion
#### Check to see if the images/ LUN mapping are created with the `targetcli ls` command
```

ravikanth-nalla-hpe replied:
Fixed.

leliasen-hpe commented on .spelling:
```diff
@@ -1068,3 +1068,11 @@ Expat-15
+enablement
+LUNs
+fileio
+cmd
+iscsi-sbps-targets-config
+targetcli
```
```suggestion
iscsi-sbps-targets-config
```
The work "cmd" should be written as command.
I believe everywhere that targetcli is used it is within "`" so I don't think it needs to be added to spelling

ravikanth-nalla-hpe replied:
Fixed.

dborman-hpe commented on operations/configuration_management/iSCSI_SBPS_Node_Personalization.md:
```diff
@@ -0,0 +1,357 @@
+In order to achieve this, CFS based Ansible plays (`csm-config`) are created and deployed in order to
+
+* provision `iSCSI targets`/`LIO services` and present the `LIO network` service via the HSN and NMN IP networks (TCP port 3260 by default), via iSCSI Portals.
+  
+* enable SBPS Marshal Agent by installing the agent rpm and starting the respective `systemd` service (`sbps-marshal`).
+  
+* enable `spire` for authentication used for `IMS/ S3` access.
+  
+* create `DNS SRV and A records` in order to discover iSCSI targets from iSCSI initiators/clients (passed as boot parameter).
+  
+* mount S3 images (`boot-images` bucket) with new, dedicated S3 user (`ISCSI-SBPS`) read only bucket policy via `s3fs`.
+  
```
The blank lines have trailing whitespace, please remove them.

ravikanth-nalla-hpe replied:
Fixed.

dborman-hpe reviewed (approved):
LGTM
